### ISSUE TITLE
Material Switch cannot be used in Sketchware Pro Redesign

### ISSUE THREAD

**Issue by TanTai97o** [Sketchware-Pro/Sketchware-Pro#1009]
M3 Switch Building with Sketchware Pro does not work, it crashes the application
But built with Android Studio or androidIDE does not have any errors
Build with AndroidIDE 

![Screenshot_20231012_123940_Material Android AIDE](https://github.com/Sketchware-Pro/Sketchware-Pro/assets/107351062/ddc300d2-6304-45e9-8796-ab129c8f2d01)

Build with Sketchware Pro Redesign 
![Screenshot_20231012_124022_Sketchware Manager](https://github.com/Sketchware-Pro/Sketchware-Pro/assets/107351062/7b3d0458-6c09-4749-8c56-753f3fc2cd69)

**Comment by appt2** [comment]
> M3 Switch Building with Sketchware Pro does not work, it crashes the application But built with Android Studio or androidAIDE does not have any errors Build with AndroidAIDE
> 
> ![Screenshot_20231012_123940_Material Android AIDE](https://user-images.githubusercontent.com/107351062/274494322-ddc300d2-6304-45e9-8796-ab129c8f2d01.png)
> 
> Build with Sketchware Pro Redesign ![Screenshot_20231012_124022_Sketchware Manager](https://user-images.githubusercontent.com/107351062/274494680-7b3d0458-6c09-4749-8c56-753f3fc2cd69.png)

Using last version All AndroidX end Try to test @JavkhlanK main is Lazy to Adding new Library to sk pro

**Comment by JavkhlanK** [comment]
if you keep spreading misinformation like a kid, we'll have to ban you from here too

**Comment by appt2** [comment]
> if you keep spreading misinformation like a kid, we'll have to ban you from here too

This information is real, some new methods have been added in the Compat switch, which cannot match the material library with it, and as a result, it gives an error. You can download and test the material switch class in sk pro.

**Comment by guhg** [comment]
Don't disrespect him though. You could've been a lot kinder in your words.
Jav is just doing his job

**Comment by guhg** [comment]
Be kind. He's busy all the time. No one is going to help you with that
attitude

**Comment by JavkhlanK** [comment]
Can you somehow prove that you're not that ninjacoder guy? Until you do, there's enough indices that you indeed are that guy.

**Comment by guhg** [comment]
Who's ninjacoder? What did he do?

**Comment by TanTai97o** [comment]
> Can you somehow prove that you're not that ninjacoder guy? Until you do, there's enough indices that you indeed are that guy.

I don't know what's going on here, so can problem be solved or not

**Comment by JavkhlanK** [comment]
It's unrelated to you, just ignore that.

**Comment by appt2** [comment]
I implemented the Switch Material class in Sketchware and found that float thumbPosition = getThumbPosition();
  This is the code that does not allow the program to be compiled and I did a little research to find that this getThumbPosition is a new feature in the Compat switch that Sketchware does not have, so it is not possible to use the material switch in Flan's Sketchware because I tested it with the Androidx library that I downloaded.  I encountered an error. I hope that the libraries will be updated

**Comment by guhg** [comment]
Yes, could you send me the project sir that you are testing for material
switch?

**Comment by JavkhlanK** [comment]
Before you use a Local library you of course have to make sure its dependencies are met. The MDC library version Sketchware Pro currently uses (1.6.1) already has access to them. If the library stil crashes, it's not our issue.

And before you whine about us "being lazy" and asking us to upgrade the library again (just like that ninjacoder guy btw), making an upgrade will force every project that uses AppCompat to also include the Kotlin stdlib which will only make projects much heavier than they already are. That means an upgrade is not happening, unless the library suddenly stops depending on Kotlin at runtime.

**Comment by appt2** [comment]
Well, why are the programs that I develop with Android IDE not heavy, only sketchware makes the program heavy, this is not logical, and we ask you to update the libraries and add some new libraries like core-kt

**Comment by JavkhlanK** [comment]
That alone shows that you have no idea what you're talking about (also just like that ninjacoder guy, what a surprise!). `*-ktx` dependencies are only needed when writing Kotlin code, and since that's basically optional, that's highly unlikely to be added as part of built-in libraries.

**Comment by appt2** [comment]
OK, we will deal with it later, I don't have enough time to discuss now

**Comment by guhg** [comment]
Man you guys are like snapping turtles...
